src/lib/ast.rs

```diff
@@ -415,6 +390,31 @@ impl NodeId {
         Ok(())
     }
 
+    /// Make one node the next (i.e. last) child of another.
+    pub fn make_child_of<T: Clone>(&self, parent: NodeId, arena: &mut Arena<T>) -> ArenaResult {
+        if self.index >= arena.size() {
+            return Err(ArenaError::NodeIdNotFound);
+        }
+        if parent.index >= arena.size() {
+            return Err(ArenaError::NodeIdNotFound);
+        }
+        self.detach(arena)?;
+        arena[*self].parent = Some(parent);
+        match arena[parent].first_child {
+            None => arena[parent].first_child = Some(*self),
+            Some(_) => (),
+        };
+        match arena[parent].last_child {
+            None => arena[parent].last_child = Some(*self),
+            Some(id) => {
+                arena[parent].last_child = Some(*self);
+                arena[id].next_sibling = Some(*self);
+                arena[*self].previous_sibling = Some(id);
+            }
+        };
+        Ok(())
```

Comment:
Because this function returns an `ArenaResult` which is either a Unit or an `ArenaError`